### PR TITLE
workshop: Update docs and support default cloud image

### PR DESCRIPTION
**Pull request by netoarmando**
Update instructions on how to build images starting with Fedora 34 using kickstart files used by Fedora to build its cloud images.

Change vagrant provisioning steps to support both prebuilt and default cloud images, removing the burden of maintaining boxes up-to-date, but also providing a way to build fresh images without external packer templates.

---
New Vagrant boxes available here: https://app.vagrantup.com/freeipa/boxes/freeipa-workshop/versions/20170517.01

### CODE REVIEW

**Review by flo-renaud — COMMENTED** [review]
Hi @netoarmando 
thanks for the PR.
There are still references to the previous repo in the curriculum (for instance README.rst mentions `$ git clone https://github.com/freeipa/freeipa-workshop.git` but the workshop was moved to https://github.com/freeipa/freeipa/ in doc/workshop).

README.rst also instructs `$ vagrant box add netoarmando/freeipa-workshop`, this should be replaced with fedora/33-cloud-base.

README.facilitators also mentions the wrong box.

**Comment by rcritten** [comment]
Would this affect someone doing a workshop at something like a conference with limited bandwidth? I think it would mean more packages need to be retrieved.

**Comment by netoarmando** [comment]
@rcritten I changed this to support both cases, using prebuilt images as default.
@flo-renaud I've updated the documentation to fix your comments and reflect the new changes.

File: doc/workshop/building.rst [hunk 1]
```diff
@@ -8,55 +8,43 @@ Requirements
...
+used by Fedora to build vagrant images:
 
-- Add the ``scripts/fedora/ipa.sh`` provisioner and copy (or
-  symlink) ``ipa.sh`` from *this* repository to ``scripts/fedora``.
-  This script installs the FreeIPA packages and creates other files
-  required for the workshop.
+- Clone the repo and checkout latest Fedora release branch::
 
+  $ git clone https://pagure.io/fedora-kickstarts.git
+  $ git checkout f34
```
Comment (flo-renaud) [inline doc/workshop/building.rst:24]: The step "cd fedora-kickstarts" is missing before git checkout

File: doc/workshop/building.rst [hunk 2]
```diff
@@ -8,55 +8,43 @@ Requirements
...
-The output box is a gzip-compressed tarball.  Unfortunately, the VM
-image it contains is not sparse and will waste a lot of space (and
-time) when Vagrant unpacks and imports the image.  Therefore we
-unpack, sparsify and repack the box::
+Build the images::
 
-  mkdir box && cd box && tar -xf ../fedora-28-x86_64-libvirt.box
-  virt-sparsify --in-place box.img
-  tar -czf ../fedora-28-x86_64-libvirt.box * && cd .. && rm -rf box
+  $BIN_PACKER build packer-template-fedora.json
```
Comment (flo-renaud) [inline doc/workshop/building.rst:41]: "cd $FREEIPA_REPO/doc/workshop" is missing

**Review by flo-renaud — COMMENTED** [review]
Hi @netoarmando 
please find additional inline comments

File: doc/workshop/building.rst (hunk 1 above)
Comment (netoarmando) [reply doc/workshop/building.rst:25]: Fixed.

File: doc/workshop/building.rst (hunk 2 above)
Comment (netoarmando) [reply doc/workshop/building.rst]: Fixed.

**Comment by flo-renaud** [comment]
Re-running `test_authconfig` as the test timed out
I had a look at the latest changes, LGTM. ACK for when PRCI completes

**Comment by flo-renaud** [comment]
master:

* 584952babffb62346802d43df8bb46ed68897f15 workshop: Update docs and support default cloud image

**Pull request closed without merging**